lib/active_graph/node/query/query_proxy_methods_of_mass_updating.rb

```diff
@@ -49,31 +49,20 @@ def delete_all_rels
         # Deletes the relationships between all nodes for the last step in the QueryProxy chain and replaces them with relationships to the given nodes.
         # Executed in the database, callbacks will not be run.
         def replace_with(node_or_nodes)
-          node_hash = idify_hash(node_or_nodes)
+          node_or_nodes = Array(node_or_nodes).map { |arg| arg.is_a?(ActiveGraph::Node) ? arg : @model.find(arg) }
```

Comment:
```suggestion
          nodes = Array(node_or_nodes).map { |arg| arg.is_a?(ActiveGraph::Node) ? arg : @model.find(arg) }
```
This is guaranteed to be an array